q2_dada2/plugin_setup.py

```diff
@@ -212,3 +212,85 @@
     description=('This method denoises paired-end sequences, dereplicates '
                  'them, and filters chimeras.')
 )
+
+
+plugin.methods.register_function(
+    function=q2_dada2.denoise_pyro,
+    inputs={'demultiplexed_seqs': SampleData[SequencesWithQuality]},
+    parameters={'trunc_len': qiime2.plugin.Int,
+                'trim_left': qiime2.plugin.Int,
+                'max_ee': qiime2.plugin.Float,
+                'trunc_q': qiime2.plugin.Int,
+                'max_len': qiime2.plugin.Int,
+                'chimera_method': qiime2.plugin.Str %
+                qiime2.plugin.Choices(_CHIM_OPT),
+                'min_fold_parent_over_abundance': qiime2.plugin.Float,
+                'n_threads': qiime2.plugin.Int,
+                'n_reads_learn': qiime2.plugin.Int,
+                'hashed_feature_ids': qiime2.plugin.Bool},
+    outputs=[('table', FeatureTable[Frequency]),
+             ('representative_sequences', FeatureData[Sequence])],
+    input_descriptions={
+        'demultiplexed_seqs': 'The single-end demultiplexed pyrosequencing '
+                              'sequences (e.g. 454, IonTorrent) to be '
+                              'denoised.'
+    },
+    parameter_descriptions={
+        'trunc_len': 'Position at which sequences should be truncated due to '
+                     'decrease in quality. This truncates the 3\' end of the '
+                     'of the input sequences, which will be the bases that '
+                     'were sequenced in the last cycles. Reads that are '
+                     'shorter than this value will be discarded. If 0 is '
+                     'provided, no truncation or length filtering will be '
+                     'performed',
+        'trim_left': 'Position at which sequences should be trimmed due to '
+                     'low quality. This trims the 5\' end of the '
+                     'of the input sequences, which will be the bases that '
+                     'were sequenced in the first cycles.',
+        'max_ee': 'Reads with number of expected errors higher than this '
+                  'value will be discarded.',
+        'trunc_q': 'Reads are truncated at the first instance of a quality '
+                   'score less than or equal to this value. If the resulting '
+                   'read is then shorter than `trunc_len`, it is discarded.',
+        'max_len': 'Remove raw reads which are longer than this value. If 0 '
```

Comment:
It's not clear what _raw reads_ means here - could this just be _reads_ or _reads prior to trimming or truncation_